tests/_serialization/test_simple_properties.py

```diff
@@ -50,8 +51,23 @@ def test_invalid_deserialization_type(prop_type, serialized):
         prop.deserialize(serialized)
 
 
+@pytest.mark.parametrize('prop_type,serialized', INVALID_DESERIALIZATION_TYPES)
+def test_invalid_deserialization_type_with_base_class(prop_type, serialized):
+    class BaseTest:
+        pass
+
+    prop = prop_type()
+    prop.serialization_path = 'ser_path'
+    with pytest.raises(ValueError) as excinfo:
+        prop.deserialize(serialized, base_class=BaseTest().__class__)
+
+    # Check that the exception includes the calling class name and argument
+    if not isinstance(prop, UUID):
+        assert 'BaseTest:ser_path' in str(excinfo.value)
+
```

Comment:
Could we add another test here that tries to deserialize an invalid serialization of an existing module (say, a dataset) and demos that the error is what we expect?

Reply:
No worries, I just added another test that uses a Dataset object as the base_class.

It's a little awkward to supply the 'base_class' parameter but I don't think there's any way to get the class name from an attribute.  For example, if d=Dataset() and d.xyz = 123, d.xyz is just a plain Python integer and I can't find a way of getting the name of it's class.  This is why I had to include this new (optional) parameter.